src/main/java/org/kohsuke/github/GHEventPayload.java

```diff
@@ -112,11 +113,14 @@ void wrapUp(GitHub root) {
             sender.wrapUp(root);
         }
         if (repository != null) {
-            repository.root = root;
+            repository.wrap(root);
```

Comment:
note: `repository.root = root;` replaced with `repository.wrap(root);`, a more complex. This way we can inherit this action in subclasses and remove `repository.wrap(root);` line from them.